src/app/beer_garden/db/mongo/models.py

```diff
@@ -396,38 +396,46 @@ class Request(MongoModel, Document):
     logger = logging.getLogger(__name__)
 
     def pre_serialize(self):
+        """Pull any fields out of GridFS"""
         encoding = "utf-8"
-        """If string output was over 16MB it was spilled over to the GridFS storage solution"""
+
         if self.output_gridfs:
-            self.logger.debug("~~~Retrieving output from gridfs")
+            self.logger.debug("Retrieving output from GridFS")
             self.output = self.output_gridfs.read().decode(encoding)
             self.output_gridfs = None
 
         if self.parameters_gridfs:
-            self.logger.debug("~~~Retrieving parameters from gridfs")
+            self.logger.debug("Retrieving parameters from GridFS")
             self.parameters = json.loads(self.parameters_gridfs.read().decode(encoding))
             self.parameters_gridfs = None
 
     def save(self, *args, **kwargs):
+        """Save a Request
+
+        This does a couple of things:
+        - Updates fields that need to be set every time
+        - Handles the case where the Request is too big to fit into a single document
+        """
         self.updated_at = datetime.datetime.utcnow()
+
         max_size = 15 * 1_000_000
+        request_size = 0
         encoding = "utf-8"
-        parameter_size = sys.getsizeof(self.parameters)
 
-        if self.output:
-            parameter_size = sys.getsizeof(self.parameters)
-            if parameter_size > max_size:
-                self.logger.info("~~~Parameter size too big, storing in gridfs")
+        if self.parameters:
+            request_size += sys.getsizeof(self.parameters)
```

Comment:
This won't work.  getsizeof doesn't actually account for the size of the values in the dict.  See [this random blog](https://nedbatchelder.com/blog/202002/sysgetsizeof_is_not_what_you_want.html) for a decent explanation. The end result is that the check below will basically always return False, even if request_size is huge.